source/NVDAObjects/UIA/wordDocument.py

```diff
@@ -544,14 +546,26 @@ class WordDocument(UIADocumentWithTableNavigation,WordDocumentNode,WordDocumentB
 	def event_textChange(self):
 		# Ensure Braille is updated when text changes,
 		# As Microsoft Word does not fire caret events when typing text, even though the caret does move.
-		braille.handler.handleCaretMove(self)
+		# Update braille also when tethered to review, and review position
+		# if review follows caret.
+		if not eventHandler.isPendingEvents("caret, self"):
+			eventHandler.queueEvent("caret", self)
 
 	def event_UIA_notification(self, activityId=None, **kwargs):
 		# #10851: in recent Word 365 releases, UIA notification will cause NVDA to announce edit functions
 		# such as "delete back word" when Control+Backspace is pressed.
 		if activityId == "AccSN2":  # Delete activity ID
 			return
 		super(WordDocument, self).event_UIA_notification(**kwargs)
+		# Try to ensure that braille is updated when UIA is not used and
+		# ctrl-v, ctrl-x or ctrl-z is pressed.

```

Comment:
While  obviously the final decision with regard to merging this or not isn't mine, I fell very uneasy with a hacky solution like this. You essentially rely on the fact that MS Word emits a notification to assistive technology when some keyboard commands are pressed, but there is very little guarantee that this will always be the case, there is also no definitive list of situations in which Word emits UIA notification, so we cannot be sure in what case this code would execute. In #3276 @michaelDCurran proposed at least two solutions to the problem:
- Rely on MS Word COM events for caret movements if any
- Create scripts bound to the keyboard command for which we need to update caret, pass the shortcut to Word, and then refresh the braille display

Have you investigated any of these approaches? If you did, and they are not sufficient I would like to have this mentioned in the PR description along with what was not working. If you didn't then the @michaelDCurran's comment in #3276 seems a good start.

Reply:
Thanks I try to investigate this.